### PR TITLE
[WIP] Arrow Flight Server

### PR DESCRIPTION
**Pull request by FawnD2**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support compatibility protocol for Arrow Flight
...

Detailed description / Documentation draft:

...

By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Server/ArrowFlightServer.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <Common/CurrentMetrics.h>
+#include <Poco/Net/TCPServerConnection.h>
+#include <common/logger_useful.h>
+#include "IServer.h"
+#include "IRoutineServer.h"
+
+#include <arrow/flight/server.h>
```
Comment (FawnD2) [inline src/Server/ArrowFlightServer.h:9]: This header is not found during compilation although I provided some info (https://github.com/ClickHouse/ClickHouse/pull/16243/files#diff-148715d6ea0c0ea0a346af3f6bd610d010d490eca35ac6a9b408748f7ca9e3f4R336) to cmake. How to change cmakefiles to fix it?

Comment (FawnD2) [reply src/Server/ArrowFlightServer.h:9]: Server was successfully built

**Comment by FawnD2** [comment]
There is an issue with ipv6 in Arrow Flight library. I created a ticket in Arrow jira: https://issues.apache.org/jira/browse/ARROW-10475

**Comment by alexey-milovidov** [comment]
@FawnD2 Should we continue working on this PR or better to close it?

**Comment by alexey-milovidov** [comment]
No answer.

**Pull request closed without merging**